examples/AspNetCore/Program.cs

```diff
@@ -14,67 +14,131 @@
+        case "otlp":
+            options.AddOtlpExporter(otlpOptions =>
                 {
-                    builder.ClearProviders();
-                    builder.AddConsole();
-
-                    var logExporter = context.Configuration.GetValue<string>("UseLogExporter").ToLowerInvariant();
-                    switch (logExporter)
-                    {
-                        case "otlp":
-                            // Adding the OtlpExporter creates a GrpcChannel.
-                            // This switch must be set before creating a GrpcChannel when calling an insecure gRPC service.
-                            // See: https://docs.microsoft.com/aspnet/core/grpc/troubleshoot#call-insecure-grpc-services-with-net-core-client
-                            AppContext.SetSwitch("System.Net.Http.SocketsHttpHandler.Http2UnencryptedSupport", true);
-                            builder.AddOpenTelemetry(options =>
-                            {
-                                options.SetResourceBuilder(ResourceBuilder.CreateDefault().AddService(context.Configuration.GetValue<string>("Otlp:ServiceName")));
-                                options.AddOtlpExporter(otlpOptions =>
-                                 {
-                                     otlpOptions.Endpoint = new Uri(context.Configuration.GetValue<string>("Otlp:Endpoint"));
-                                 });
-                            });
-                            break;
-
-                        default:
-                            builder.AddOpenTelemetry(options =>
-                            {
-                                options.AddConsoleExporter();
-                            });
-                            break;
-                    }
-
-                    builder.Services.Configure<OpenTelemetryLoggerOptions>(opt =>
-                    {
-                        opt.IncludeScopes = true;
-                        opt.ParseStateValues = true;
-                        opt.IncludeFormattedMessage = true;
-                    });
+                    otlpOptions.Endpoint = new Uri(builder.Configuration.GetValue<string>("Otlp:Endpoint"));
                 });
+            break;
+
+        default:
+            options.AddConsoleExporter();
+
+            break;
     }
+});
+
+builder.Services.AddOpenTelemetryMetrics(options =>
+{
+    options.SetResourceBuilder(resourceBuilder)
+        .AddHttpClientInstrumentation();
+
+    var metricsExporter = builder.Configuration.GetValue<string>("UseMetricsExporter").ToLowerInvariant();
+    switch (metricsExporter)
+    {
+        case "prometheus":
+            options.AddPrometheusExporter();
+            break;
+        case "otlp":
+            options.AddOtlpExporter();
+            break;
+        default:
+            options.AddConsoleExporter();
+            break;
+    }
+});
+
+builder.Services.Configure<AspNetCoreInstrumentationOptions>(builder.Configuration.GetSection("AspNetCoreInstrumentation"));
```

Comment:
Missing options from existing example:
```// For options which can be configured from code only.
                    services.Configure<AspNetCoreInstrumentationOptions>(options =>
```

Reply:
This code fails saying the statement is aways true (req.Request.Host != null;).
 // For options which can be configured from code only.
                    services.Configure<AspNetCoreInstrumentationOptions>(options =>
                    {
                        options.Filter = (req) =>
                        {
                            return req.Request.Host != null;
                        };
                    });